pyuvdata/uvdata/uvdata.py

```diff
@@ -2606,31 +2606,230 @@ def _check_freq_spacing(self, raise_errors=True):
+                if not np.array_equal(this_spw_order, pol0_spw_order):
+                    reorder_spws = True
+
+            if reorder_channels or reorder_spws:
+                if reorder_channels:
+                    channel_order = "freq"
+                else:
+                    channel_order = None
+                if reorder_spws:
+                    # note: spw_order is an index array into spw_array
+                    spw_order = np.zeros(self.Nspws, dtype=int)
+                    if np.all(unique_pols < 0):
+                        # use more standard ordering for polarizations
+                        unique_pols = unique_pols[np.argsort(np.abs(unique_pols))]
+                    for pol_ind, pol in enumerate(unique_pols):
+                        for spw_ind, (_, spw_set) in enumerate(spw_dict.items()):
+                            this_ind = pol_ind * n_sets + spw_ind
+                            this_spw = np.array(spw_set["spws"])[spw_set["pols"] == pol]
+                            spw_order[this_ind] = np.nonzero(
+                                self.spw_array == this_spw
+                            )[0]
+                else:
+                    spw_order = None
+
+                self.reorder_freqs(channel_order=channel_order, spw_order=spw_order)
+
+                # recalculate arrays used below
+                if self.future_array_shapes:
+                    freq_array_use = self.freq_array
+                else:
+                    freq_array_use = self.freq_array[0, :]
+                first_spw_array = np.zeros_like(self.flex_spw_id_array)
+                for spw_ind, spw in enumerate(self.spw_array):
+                    these_freq_inds = np.nonzero(self.flex_spw_id_array == spw)[0]
+                    spw_inds[these_freq_inds] = spw_ind
+                    first_spw_array[these_freq_inds] = first_spw_dict[spw]
+                pol_array_check = self.flex_spw_polarization_array[spw_inds]
+
+            self.Npols = n_pols
+            self.Nspws = n_sets
+            self.Nfreqs = n_freqs
+
+            # now things are in the correct order to do a simple reshape
+            if self.future_array_shapes:
+                self.freq_array = freq_array_use[: self.Nfreqs]
+            else:
+                self.freq_array = freq_array_use[np.newaxis, : self.Nfreqs]
+            self.channel_width = self.channel_width[: self.Nfreqs]
+
+            self.polarization_array = pol_array_check[
+                np.sort(np.unique(pol_array_check, return_index=True)[1])
+            ]
+            self.flex_spw_polarization_array = None
+
+            self.spw_array = first_spw_array[
+                np.sort(np.unique(first_spw_array, return_index=True)[1])
+            ]
+            self.flex_spw_id_array = first_spw_array[: self.Nfreqs]
+            if not self.metadata_only:
+                if self.future_array_shapes:
+                    self.data_array = self.data_array.reshape(
```

Comment:
Sorry, but why do we want the `order="F"` here? Don't we want our arrays to be C-ordered when we're done?

Reply:
This controls whether the frequency axis or polarization axis changes slowest. "F" is required for the polarization to change slowest.

Reply:
Doesn't the order="F" just kinda obfuscate a transposition though? we could just manually transpose the array to be more clear about what's happening to a new reader.

Reply:
I'll add a comment explaining this.